### DISCUSSION TITLE
[IntelliSense] - autocomplete for custom class added using layer directive

### DISCUSSION THREAD

**Discussion by omerman** [tailwindlabs/tailwindcss#7259] [Ideas]
![image](https://user-images.githubusercontent.com/6780529/151667067-6afd9576-c46e-4451-b225-9aa97cae86f3.png)
I think it would be cool if by adding a custom class, the plugin will be able to auto complete for the custom classes as well.

![image](https://user-images.githubusercontent.com/6780529/151667159-51ae2f69-1a84-4c1a-be87-760d7ffe9c14.png)

Happy to hear your thoughts on the matter

Thank you for this great framework :)

**Comment by wiebekaai** [comment]
Yes please! I've been working around this using a custom plugin, but it's not as readable.

1. [Create a Tailwind plugin with components](https://github.com/tailwindlabs/tailwindcss-intellisense/issues/227#issuecomment-1269592872)
2. Adjust some settings (see snippet below, requires bradlc.vscode-tailwindcss extension)

`.vscode/settings.json`
```
"tailwindCSS.includeLanguages": {
  "javascript": "javascript"  
},
"[javascript]": {
  "tailwindCSS.experimental.classRegex": [
    "'@apply([^']*)':"
  ],
},
```

**Comment by sonpapadi8** [comment]
What a fabulous post this has been. Never seen this kind of useful post. I am grateful to you and expect more number of posts like these. Thank you very much. [Casey's Feedback](https://www.survey.onl/caseys-feedback/)
